src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs

```diff
@@ -140,46 +140,30 @@ private DirectoryInfo GetTestDestinationDirectory(string callingMethod, string i
 
         private IEnumerable<FileInfo> LoadInventory(FileInfo file)
         {
+            file.Refresh();
             if (!file.Exists)
             {
-                return Enumerable.Empty<FileInfo>();
+                throw new InvalidOperationException("Inventory file should exist.");
             }
 
             var inventory = new List<FileInfo>();
-
-            var lines = file.OpenText();
-
-            while (lines.Peek() > 0)
+            foreach (var p in File.ReadAllLines(file.FullName))
             {
-                inventory.Add(new FileInfo(lines.ReadLine()));
+                inventory.Add(new FileInfo(p));
             }
 
             return inventory;
         }
 
         private void SaveInventory(FileInfo file, IEnumerable<FileInfo> inventory)
         {
-            FileUtility.ReplaceWithLock(
-                filePath =>
-                {
-                    if (!_dataDirectory.Exists)
-                    {
-                        _dataDirectory.Create();
-                    }
+            _dataDirectory.Refresh();
```

Comment:
idempotency seems appropriate. This exists based on the stipulation that identical results are expected. 

@jonsequitur what pattern would you recommend to express the intent of the handler?